src/main/java/com/example/onlinebookstore/repository/AbstractRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.example.onlinebookstore.repository;
+
+import jakarta.persistence.EntityManagerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public abstract class AbstractRepository {
+    protected EntityManagerFactory manager;
```

Comment:
I`m not ssure you need this abstract class. But, maybe it can be useful in future)